mamba/mamba/mamba_shell_init.py

```diff
@@ -81,12 +81,18 @@ def shell_init(args):
         return initialize_dev(shell)
 
     else:
-        for_user = args.user
-        if not (args.install and args.user and args.system):
-            for_user = True
-        # no_user removed in Conda 23.1.0, see https://github.com/mamba-org/mamba/issues/2248
-        if hasattr(args, "no_user") and args.no_user:
-            for_user = False
+        if hasattr(args, "no_user"):
+            # this seems to be conda < 23.1.0
+            # Here, we use the old behavior even though the --no-user flag
+            # doesn't behave as desired, see:
+            # https://github.com/conda/conda/issues/11948
+            for_user = args.user
+            if not (args.install and args.user and args.system):
+                for_user = True
+            if args.no_user:
+                for_user = False
```

Comment:
Okay, this is the block of code before #2249.  So we'll behave as before if `args.no_user` is present but things will work properly (including the `--no-user` flag for newer versions where the argument is no longer there.